compiler/rustc_hir_analysis/src/hir_ty_lowering/dyn_compatibility.rs

```diff
@@ -352,6 +352,17 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                     b.projection_term = replace_dummy_self_with_error(tcx, b.projection_term, guar);
                 }
 
+                if let Some(term_ty) = &b.term.as_type() {
```

Comment:
You can combine this with the `let references_self` above. You should probably just chain this like `b.projection_term.args.iter().skip(1).chain([b.term.into()])` to make sure we check the term for `Self` as well.

Reply:
I changed it to an error now.